### PR TITLE
GH-15164: Add new bloom filter example for current spec

### PR DESCRIPTION
**Pull request by mapleFU**
This patch is part of https://github.com/apache/arrow/pull/33776 .
The original `bloom_filter.bin` uses the original Murmur3-based format added in PARQUET-41.
That bloom filter format was later replaced with a xxHash-based format.

The BF is generated by parquet-mr. It has 4 strings hashed by xxhash:
```
"hello", "parquet", "bloom", "filter"
```

### CODE REVIEW

**Review by wjones127 — APPROVED** [review]

**Comment by mapleFU** [comment]
@pitrou Would you mind to take a look here?

**Comment by emkornfield** [comment]
its hard to tell from the diff, but if this is being used in a unit test already and we are changing it I would prefer to create a new file "e.g. bloom_filter.xxhash.bin" and the remove the old one separately.

**Comment by mapleFU** [comment]
@emkornfield Thanks! I've done it

**Comment by mapleFU** [comment]
(Sorry for mistouch...I don't know how to revoke review...)

File: data/README.md [hunk 1]
```diff
@@ -38,6 +38,8 @@
 | datapage_v1-snappy-compressed-checksum.parquet | compressed INT32 columns in v1 data pages with a matching CRC          |
 | datapage_v1-corrupt-checksum.parquet           | uncompressed INT32 columns in v1 data pages with a mismatching CRC     |
 | overflow_i16_page_cnt.parquet                  | row group with more than INT16_MAX pages                   |
+| bloom_filter.bin                               | deprecated bloom filter binary with binary header and murmur3 hashing |
+| bloom_filter.xxhash.bin                        | bloom filter binary with thrift header and xxhash hashing    |
```
Comment (pitrou) [inline data/README.md:42]: Does "binary" here mean these are not complete Parquet files?

File: data/README.md [hunk 2]
```diff
@@ -38,6 +38,8 @@
 | datapage_v1-snappy-compressed-checksum.parquet | compressed INT32 columns in v1 data pages with a matching CRC          |
 | datapage_v1-corrupt-checksum.parquet           | uncompressed INT32 columns in v1 data pages with a mismatching CRC     |
 | overflow_i16_page_cnt.parquet                  | row group with more than INT16_MAX pages                   |
+| bloom_filter.bin                               | deprecated bloom filter binary with binary header and murmur3 hashing |
```
Comment (pitrou) [inline data/README.md:41]: What do you mean with "binary header"? Doesn't the [murmur3 bloom filter format](https://github.com/apache/parquet-format/commit/54839ad5e04314c944fed8aa4bc6cf15e4a58698) already use a Thrift header (`BloomFilterPageHeader`)?

File: data/README.md (hunk 1 above)
Comment (mapleFU) [reply data/README.md:42]: Yes, there is only bloom filter, no complete parquet file

File: data/README.md (hunk 2 above)
Comment (mapleFU) [reply data/README.md:41]: Hi pitrou, I'm also confuse with this. But original code, with works well with `bloom_filter.bin`, uses binary parsing

**Comment by pitrou** [comment]
Thanks a lot @mapleFU ! I've updated the description and will merge this PR.

**Pull request merged**